pkg/query-service/main.go

```diff
@@ -39,18 +38,11 @@ func main() {
 	var preferDelta bool
 	var preferSpanMetrics bool
 
-	var maxIdleConns int
-	var maxOpenConns int
-	var dialTimeout time.Duration
-
 	flag.StringVar(&promConfigPath, "config", "./config/prometheus.yml", "(prometheus config to read metrics)")
 	flag.StringVar(&skipTopLvlOpsPath, "skip-top-level-ops", "", "(config file to skip top level operations)")
 	flag.BoolVar(&disableRules, "rules.disable", false, "(disable rule evaluation)")
 	flag.BoolVar(&preferDelta, "prefer-delta", false, "(prefer delta over cumulative metrics)")
 	flag.BoolVar(&preferSpanMetrics, "prefer-span-metrics", false, "(prefer span metrics for service level metrics)")
-	flag.IntVar(&maxIdleConns, "max-idle-conns", 50, "(number of connections to maintain in the pool.)")
-	flag.IntVar(&maxOpenConns, "max-open-conns", 100, "(max connections for use at any time.)")
-	flag.DurationVar(&dialTimeout, "dial-timeout", 5*time.Second, "(the maximum time to establish a connection.)")
```

Comment:
The default values used here should be used in `defaultConnector` parse logic if they are not set.

Reply:
There doesn't seem to be any defined "Defaults" in the clickhouse library, although there is some logic checks in the setDefaults function on the clickhouse_options.go file that do basic out of range logic checks.

I've followed this semantic by setting the flag defaults to 0 of which will trigger the clickhouse defaults to be set.

Reply:
No, we don't want the ClickHouse-go defaults. They are really small and throw error for non-trival dashboards. And when the number of alerts is high. We have found the current settings to be effective and they should be the defaults.

Reply:
Ok,

I've added in logic to use the Flags only if the DSN does not explicitly set them. This will ensure the DSN configuration will be respected first and then the flags thereafter.